File: typescript/src/createUser/getUserData.ts
import axios from 'axios';
import { GithubUserResponseData } from './GithubResponseTypes';

export default async (accessToken: string): Promise<GithubUserResponseData> => {
  const headers = { headers: { Authorization: `token ${accessToken}` } };

  try {
    const { data } = await axios.get('https://api.github.com/user', headers);
    return Promise.resolve(data);
  } catch (e) {
    return Promise.reject('Unauthorized');
  }
}

File: rails-tdd/app/services/user/create/get_user_data.rb
module User
  module Create
    class GetUserData
      def self.call(access_token)
        p access_token
        res = HTTP[:Authorization => "token #{access_token}"]
          .get('https://api.github.com/user')

        parsedBody = JSON.parse res.body.to_s

        if res.code != 200 
          raise "Could not get user data"
        end

        if parsedBody.has_key? :error
          raise "Could not get user data"
        end

        parsedBody 
      end
    end
  end
end